templates/jutsu_page.php

```diff
@@ -1,126 +1,303 @@
+                }
+                if (!$("#jutsu_filter_genjutsu").hasClass('active')) {
+                    jutsu.filter('[data-jutsu_type="Genjutsu"]').addClass('hidden');
+                }
+            }
+            if (!$("#jutsu_filter_damage").is(':checked')) {
+                jutsu.filter('[data-jutsu_effect*="None"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect*="Residual Damage"]').addClass('hidden');
+            }
+            if (!$("#jutsu_filter_buff").is(':checked')) {
+                jutsu.filter('[data-jutsu_effect="Taijutsu Boost"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Ninjutsu Boost"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Genjutsu Boost"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Speed Boost"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Cast Speed Boost"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Barrier"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Release Genjutsu"]').addClass('hidden');
+            }
+            if (!$("#jutsu_filter_debuff").is(':checked')) {
+                jutsu.filter('[data-jutsu_effect="Taijutsu Nerf"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Ninjutsu Nerf"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Genjutsu Nerf"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Speed Nerf"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Cast Speed Nerf"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Willpower Nerf"]').addClass('hidden');
+                jutsu.filter('[data-jutsu_effect="Intelligence Nerf"]').addClass('hidden');
+            }
+        }
+        // Drag and drop functionality
+        $(".jutsu_block_table").on('dragstart', function (e) {
+            e.originalEvent.dataTransfer.effectAllowed = 'move';
+            e.originalEvent.dataTransfer.dropEffect = 'move';
+            e.originalEvent.dataTransfer.setData('text/html', $(this).attr("data-jutsu_select"));
+        });
+        $(".jutsu_select").on('dragenter', function (e) {
+            e.preventDefault();
+            this.parentElement.classList.add('over');
+        });
+        $(".jutsu_select").on('dragover', function (e) {
+            e.preventDefault();
+        });
+        $(".jutsu_select").on('dragleave', function () {
+            this.parentElement.classList.remove('over');
+        });
+        $(".jutsu_select").on('drop', function (e) {
+            e.preventDefault();
+            this.parentElement.classList.remove('over');
+            console.log(e.originalEvent.dataTransfer.getData('text/html'));
+            $(this).val((e.originalEvent.dataTransfer.getData('text/html')));
+        });
+    });
```

Comment:
At the point where we have this much display logic ideally would love to see it in a React component or a similar functional pattern where:
- You have a function called `render(currentState)`
- It contains all the display logic to show the whole page based off of whatever `currentState` is
- Then whenever you have an event listener or whatnot you can just update `currentState` and call `render()`

Not a blocker as I prefer to get the UX improvement for users, but directionally would like to move away from large blocks of jQuery manually mutating DOM elements in event handlers as it gets a bit unwieldy to reason around